docs/guides/latest/workloads/statefulset.md

```diff
@@ -0,0 +1,809 @@
+    Environment:
+      NODE_NAME:   (v1:spec.nodeName)
+      POD_NAME:    (v1:metadata.name)
+    Mounts:
+      /etc/stash/repository/secret from stash-secret-volume (rw)
+      /source/data from source-data (rw)
+      /tmp from tmp-dir (rw)
+  Containers:
+   busybox:
+    Image:      busybox
+    Port:       <none>
+    Host Port:  <none>
+    Command:
+      sleep
+      3600
+    Environment:  <none>
+    Mounts:
+      /source/data from source-data (rw)
+  Volumes:
+   tmp-dir:
+    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
+    Medium:     
+    SizeLimit:  <unset>
+   stash-podinfo:
+    Type:  DownwardAPI (a volume populated by information about the pod)
+    Items:
+      metadata.labels -> labels
+   stash-secret-volume:
+    Type:        Secret (a volume populated by a Secret)
+    SecretName:  gcs-secret
+    Optional:    false
+Volume Claims:
+  Name:          source-data
+  StorageClass:  standard
+  Labels:        <none>
+  Annotations:   <none>
+  Capacity:      1Gi
+  Access Modes:  [ReadWriteOnce]
+...
+```
+
+Notice the `Init-Containers` section. We can see that init-container `stash-init` has been injected which is running `restore` command.
+
+**Wait for RestoreSession to Succeeded:**
+
+Run the following command to watch RestoreSession phase,
+
+```console
+$ watch -n 3 kubectl get restoresession -n demo
+Every 5.0s: kubectl get restoresession -n demo               suaas-appscode: Tue Jun 25 18:27:30 2019
+
+NAME         REPOSITORY-NAME   PHASE       AGE
+ss-restore   gcs-repo          Running     2m15s
+ss-restore   gcs-repo          Succeeded   4m31s
+```
+
+So, we can see from the output of the above command that the restore process succeeded.
+
+**Verify Restored Data:**
+
+In this section, we will verify that the desired data has been restored successfully.
+
+At first, check if the `stash-recovered` StatefulSet's pod has gone into `Running` state by the following command,
+
+```console
+$ kubectl get pod -n demo
+NAME                READY   STATUS    RESTARTS   AGE
+stash-recovered-0   1/1     Running   0          10m
+stash-recovered-1   1/1     Running   0          11m
+stash-recovered-2   1/1     Running   0          12m
+```
+
+Verify that the sample data has been restored in `/source/data` directory of the `stash-recovered` StatefulSet's pod using the following command,
+
+```console
+$ kubectl exec -n demo stash-recovered-0 -- ls -R /source/data
+/source/data:
+stash-demo-0.txt
+$ kubectl exec -n demo stash-recovered-1 -- ls -R /source/data
+/source/data:
+stash-demo-1.txt
+$ kubectl exec -n demo stash-recovered-2 -- ls -R /source/data
+/source/data:
+stash-demo-2.txt
+```
+
+### Rule Based Restore Process
+
+Generally, Stash runs restore process in all pod's of a StatefulSet. It also provide you what data will be restored into which pod by configuring the `spec.rules`in `BackupSession` crd. You can learn more details from [here](docs/concepts/crds/restoresession.md#specrules)
```

Comment:
Generally, Stash restores data in individual replica from the backup of respective replica of the original StatefulSet. That means, backed up data of `pod-0` of original StatefulSet will be restored in `pod-0` of new StatefulSet and so on. However, you can customize this behavior through `spec.rules` section of RestoreSession object. This is particularly helpful when your restored StatefulSet has different replica than the original StatefulSet. You can control which data will be restored in the excessed or reduced replicas.